src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -163,6 +185,51 @@ void snc_quota_manager::record_response_tp(
     _shard_quota.out.use(request_size, now);
 }
 
+std::chrono::milliseconds
+snc_quota_manager::get_quota_balancer_node_period() const {
+    const auto v = _kafka_quota_balancer_node_period();
+    // zero period in config means do not run balancer
+    if (v == std::chrono::milliseconds::zero()) {
+        return std::chrono::duration_cast<std::chrono::milliseconds>(
+          ss::steady_clock_type::duration::max() / 2);
```

Comment:
thanks for the explanation. i'm not sure its reasonable to think someone reading this would derive all of this information. please add this information into the code or change the code so that the semantics are clear.